packages/runtime-core/src/renderer.ts

```diff
@@ -2499,7 +2499,8 @@ function baseCreateRenderer(
   const getNextHostNode: NextFn = vnode => {
     if (vnode.shapeFlag & ShapeFlags.COMPONENT) {
       if ((vnode.type as ConcreteComponent).__vapor) {
-        return hostNextSibling((vnode.component! as any).block)
+        const next = hostNextSibling((vnode.component! as any).block)
```

Comment:
It should be `vnode.anchor`
I have already fixed it at [#12962](https://github.com/vuejs/core/pull/12962)https://github.com/vuejs/core/blob/98e1dd1156b6635b9d670411bce93be34ee107fb/packages/runtime-core/src/renderer.ts#L2514